packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFUserContentControllerHostApi.m

```diff
@@ -0,0 +1,80 @@
+
+@interface FWFUserContentControllerHostApiImpl ()
+@property(nonatomic) FWFInstanceManager *instanceManager;
+@end
+
+@implementation FWFUserContentControllerHostApiImpl
+- (instancetype)initWithInstanceManager:(FWFInstanceManager *)instanceManager {
+  self = [self init];
+  if (self) {
+    _instanceManager = instanceManager;
+  }
+  return self;
+}
+
+- (WKUserContentController *)userContentControllerForIdentifier:(NSNumber *)instanceId {
+  return (WKUserContentController *)[self.instanceManager
+      instanceForIdentifier:instanceId.longValue];
+}
+
+- (void)createFromWebViewConfigurationWithIdentifier:(nonnull NSNumber *)instanceId
+                             configurationIdentifier:(nonnull NSNumber *)configurationInstanceId
+                                               error:(FlutterError *_Nullable *_Nonnull)error {
+  WKWebViewConfiguration *configuration = (WKWebViewConfiguration *)[self.instanceManager
+      instanceForIdentifier:configurationInstanceId.longValue];
+  [self.instanceManager addInstance:configuration.userContentController
+                     withIdentifier:instanceId.longValue];
+}
+
+- (void)addScriptMessageHandlerForControllerWithIdentifier:(nonnull NSNumber *)instanceId
+                                         handlerIdentifier:(nonnull NSNumber *)handler
+                                                    ofName:(nonnull NSString *)name
+                                                     error:
+                                                         (FlutterError *_Nullable *_Nonnull)error {
+  [[self userContentControllerForIdentifier:instanceId]
+      addScriptMessageHandler:(id<WKScriptMessageHandler>)[self.instanceManager
+                                  instanceForIdentifier:handler.longValue]
+                         name:name];
+}
+
+- (void)removeScriptMessageHandlerForControllerWithIdentifier:(nonnull NSNumber *)instanceId
+                                                         name:(nonnull NSString *)name
+                                                        error:(FlutterError *_Nullable *_Nonnull)
+                                                                  error {
+  [[self userContentControllerForIdentifier:instanceId] removeScriptMessageHandlerForName:name];
+}
+
+- (void)removeAllScriptMessageHandlersForControllerWithIdentifier:(nonnull NSNumber *)instanceId
+                                                            error:
+                                                                (FlutterError *_Nullable *_Nonnull)
+                                                                    error {
+  if (@available(iOS 14.0, *)) {
+    [[self userContentControllerForIdentifier:instanceId] removeAllScriptMessageHandlers];
+  } else {
+    *error = [FlutterError
+        errorWithCode:@"FWFUnsupportedVersionError"
+              message:@"removeAllScriptMessageHandlers is only supported on versions 14+."
+              details:nil];
```

Comment:
Not blocking getting this new implementation completed and deployed, but we should audit the new webview implementations as a follow-up later to do the error mapping from `PlatformExceptions` that's described in the wiki now. E.g., we should have a specific, platform-neutral, package-specific Dart error+code for operations that aren't supported on the current platform.